### PR TITLE
Add protocol selection to Radius serveur configuration

### PR DESCRIPTION
**Pull request by ubawurinna**

### CODE REVIEW

**Comment by pfSense-git-updates** [comment]
Before this pull request can be accepted you must first sign a CLA as described at https://www.pfsense.org/about-pfsense/#cla. Please read for more details.

**Comment by sbeaver-netgate** [comment]
Thanks for this work. Since this it is a new feature we cannot merge it into 2.3 Beta now. Please save the code and resubmit for 2.3.1 after 2.3 is released.

**Comment by rbgarga** [comment]
Could you please rebase your fork?

File: src/etc/inc/auth.inc [hunk 1]
```diff
@@ -1362,29 +1362,65 @@ function ldap_backed($username, $passwd, $authcfg) {
 	return true;
 }
 
-function radius_backed($username, $passwd, $authcfg, &$attributes = array()) {
+function radius_backed($username, $password, $authcfg, &$attributes = array()) {
 	global $debug, $config;
 	$ret = false;
 
 	require_once("radius.inc");
+	require_once("CHAP.inc");
```
Comment (rbgarga) [inline src/etc/inc/auth.inc]: CHAP.inc is not available in the tree anymore, you should require Crypt/CHAP.php here

File: src/etc/inc/auth.inc [hunk 2]
```diff
@@ -1396,29 +1396,65 @@ function ldap_backed($username, $passwd, $authcfg) {
...
 
-	$rauth = new Auth_RADIUS_PAP($username, $passwd);
 	if ($authcfg) {
 		$radiusservers = array();
 		$radiusservers[0]['ipaddr'] = $authcfg['host'];
 		$radiusservers[0]['port'] = $authcfg['radius_auth_port'];
 		$radiusservers[0]['sharedsecret'] = $authcfg['radius_secret'];
 		$radiusservers[0]['timeout'] = $authcfg['radius_timeout'];
+
+		$radius_protocol = $authcfg['radius_protocol'];
```
Comment (jim-p) [inline src/etc/inc/auth.inc]: Assume the default value here instead of using upgrade code.

**Comment by jim-p** [comment]
One additional request: For tracking purposes, please open a feature request ticket on https://redmine.pfsense.org/projects/pfsense that briefly describes the change (like your PR summary) and links back to this PR. Set the target to 2.4.

**Comment by rbgarga** [comment]
Committed a single diff instead of merging PR. Thanks!

**Pull request closed without merging**